Signal/src/ViewControllers/ContactsViewHelper.m

```diff
@@ -79,18 +79,18 @@ - (void)dealloc
 
 - (void)signalAccountsDidChange:(NSNotification *)notification
 {
-    dispatch_async(dispatch_get_main_queue(), ^{
-        [self updateContacts];
-    });
+    OWSAssert([NSThread isMainThread]);
```

Comment:
What I like about OWSAssert() is that it completely compiles out in production. 

We could probably use preprocessor directives to achieve the same thing with AssertIsOnMainThread(), while remaining swift-compatible.  I can't figure out how to detect if when an objective-c header is being used from swift.